### PR TITLE
Improve UX when working with custom AMIs by having AMI Family explicitly specified

### PR DESCRIPTION
**Pull request by TiberiuGC**
### Description

There are various validations eksctl performs when a custom AMI is being set (e.g. `overrideBootstrapCommand` field is not supported when AMI Family is Bottlerocket). However, if a user does not explicitly set the AMI Family via config file or CLI flags, this will result in eksctl being unable to properly perform the validations, hence having inconsistent behaviour. This PR aims to fix this problem.

Closes https://github.com/weaveworks/eksctl/issues/5756
Closes https://github.com/weaveworks/eksctl/issues/6029

### Checklist
- [x] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: pkg/apis/eksctl.io/v1alpha5/defaults.go [hunk 1]
```diff
@@ -120,8 +120,11 @@ func SetNodeGroupDefaults(ng *NodeGroup, meta *ClusterMeta, controlPlaneOnOutpos
 // SetManagedNodeGroupDefaults sets default values for a ManagedNodeGroup
 func SetManagedNodeGroupDefaults(ng *ManagedNodeGroup, meta *ClusterMeta, controlPlaneOnOutposts bool) {
 	setNodeGroupBaseDefaults(ng.NodeGroupBase, meta)
-	if ng.AMIFamily == "" {
-		ng.AMIFamily = NodeImageFamilyAmazonLinux2
+
+	// When using custom AMIs, we want the user to explicitly specify AMI family.
+	// Thus, we only setup default AMI family when no custom AMI is being used.
+	if ng.AMIFamily == "" && ng.AMI == "" {
+		ng.AMIFamily = DefaultNodeImageFamily
```
Comment (Himangini) [inline pkg/apis/eksctl.io/v1alpha5/defaults.go:127]: Shouldn't these changes be for line 103?

Comment (Himangini) [inline pkg/apis/eksctl.io/v1alpha5/defaults.go:127]: Shouldn't NodeImageFamilyAmazonLinux2 be the default for MNGs?
```suggestion
	if ng.AMIFamily == "" && ng.AMI == "" {
		ng.AMIFamily = NodeImageFamilyAmazonLinux2
```

**Review by Himangini — COMMENTED** [review]
see comments below

File: pkg/apis/eksctl.io/v1alpha5/defaults.go (hunk 1 above)
Comment (TiberiuGC) [reply pkg/apis/eksctl.io/v1alpha5/defaults.go:127]: Self managed nodegroup default values are being set **_after_** validations, thus the validations are being done directly on user input and we don't need these changes there.
image

For EKS managed nodegroups, the default values are being set _**before**_ validations. Thus, we don't want to set the default AMI Family before validating, as we want the user to do it explicitly.
image

Comment (TiberiuGC) [reply pkg/apis/eksctl.io/v1alpha5/defaults.go:127]: That's ok, I'm fine with reverting that line. I missed the error indeed.

Comment (TiberiuGC) [reply pkg/apis/eksctl.io/v1alpha5/defaults.go]: https://github.com/weaveworks/eksctl/pull/6243/commits/fd9d041e2c7407ea17762bba38096fa432cf87e8

File: pkg/apis/eksctl.io/v1alpha5/validation_test.go [hunk 2]
```diff
@@ -177,6 +185,7 @@ var _ = Describe("ClusterConfig validation", func() {
 			ng0 := cfg.NewNodeGroup()
 			ng0.Name = "node-group"
 			ng0.AMI = "ami-1234"
+			ng0.AMIFamily = api.DefaultNodeImageFamily
```
Comment (Himangini) [inline pkg/apis/eksctl.io/v1alpha5/validation_test.go:188]: should we make the vars here consistent as well?

File: pkg/cfn/builder/managed_launch_template_test.go [hunk 3]
```diff
@@ -101,6 +101,7 @@ var _ = Describe("ManagedNodeGroup builder", func() {
 					Name:         "custom-ami",
 					InstanceType: "m5.xlarge",
 					AMI:          "ami-custom",
+					AMIFamily:    api.DefaultNodeImageFamily,
```
Comment (Himangini) [inline pkg/cfn/builder/managed_launch_template_test.go:104]: same here

File: userdocs/src/usage/custom-ami-support.md [hunk 4]
```diff
@@ -79,6 +79,9 @@ managedNodeGroups:
 
 The `--node-ami-family` flag can also be used with `eksctl create nodegroup`.
 
+!!! note
+  To ensure consistent behaviour through various validations, eksctl requires AMI Family to be explicitly set via config file or via `--node-ami-family` CLI flag, whenever working with a custom AMI.
+
```
Comment (Himangini) [inline userdocs/src/usage/custom-ami-support.md:84]:
```suggestion
The `--node-ami-family` flag can also be used with `eksctl create nodegroup`. `eksctl` requires AMI Family to be explicitly set via config file or via `--node-ami-family` CLI flag, whenever working with a custom AMI.

```

File: pkg/apis/eksctl.io/v1alpha5/validation_test.go (hunk 2 above)
Comment (TiberiuGC) [reply pkg/apis/eksctl.io/v1alpha5/validation_test.go:188]:
```suggestion
			ng0.AMIFamily = api.NodeImageFamilyAmazonLinux2
```

File: pkg/cfn/builder/managed_launch_template_test.go (hunk 3 above)
Comment (TiberiuGC) [reply pkg/cfn/builder/managed_launch_template_test.go:104]:
```suggestion
					AMIFamily:    api.NodeImageFamilyAmazonLinux2,
```

**Review by Himangini — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**